webhook_server/libs/handlers/runner_handler.py

```
@@ -735,6 +735,79 @@ async def is_branch_exists(self, branch: str) -> bool:
+            self.logger.debug(f"{self.log_prefix} No Signed-off-by in original PR commits, skipping author restore")
+            return False
+
+        # Check if the cherry-picked commit author already matches
+        rc, current_author_email, _ = await run_command(
+            command=f"{git_cmd} log -1 --format=%ae",
+            log_prefix=self.log_prefix,
+            redact_secrets=[github_token],
+            mask_sensitive=self.github_webhook.mask_sensitive,
+        )
+        if not rc:
+            self.logger.warning(f"{self.log_prefix} Could not read current author email, proceeding with author amend")
```

Ruff (0.15.10)

[warning] Logging statement uses f-string

(G004)